tools/codegen/pkg/manifestmerge/generator.go

```diff
@@ -463,7 +471,11 @@ func getCRDsToRender(resultingCRDs []crdForFeatureSet, crdFilenamePattern, outpu
 			continue
 		}
 
-		crdFilename := strings.ReplaceAll(crdFilenamePattern, "MARKERS", fmt.Sprintf("-%s", utils.ClusterProfileToShortName(clusterProfile)))
+		clusterProfileShortName, err := utils.ClusterProfileToShortName(clusterProfile)
+		if err != nil {
+			panic(fmt.Sprintf("unrecognized clusterprofile name %q: %w", clusterProfile, err))
```

Comment:
Why is this a panic rather than failing gracefully? Is this user in put or a code problem?